plugins/inputs/consul_metrics/consul_metrics.go

```diff
@@ -0,0 +1,195 @@
+package consul_metrics
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// Consul_metrics configuration object
+type ConsulMetrics struct {
+	URL string `toml:"url"`
+
+	TokenFile string `toml:"token_file"`
+	Token     string `toml:"token"`
+
+	Timeout config.Duration `toml:"response_timeout"`
```

Comment:
Yeah I meant the user-facing option to be called `timeout`. :-) I'm ok with keeping it as is if you plan to introduce more timeouts...

Reply:
Ops, didn't get that..I'm going to fix this